### ISSUE TITLE
insert_first and insert_last does not work on empty doubly-linked list.

### ISSUE THREAD

**Issue by ishwar00** [cs3110/textbook-solutions#2]
https://github.com/cs3110/textbook-solutions/blob/80d164a4c0121cf6559b0880277ab255bc44c72a/chapter7.ml#L172-L181

This solution does not address the case of _empty doubly-linked list_ ```{first=None; last=None}```.

**producing incorrect result**:

```ml
# let dlst = empty ();;
val dlst : '_weak6 dlist = {first = None; last = None}
```

```ml
# insert_first dlst 9;;
- : unit = ()
```

```ml
# dlst;;
- : int dlist =
{first = Some {next = None; prev = None; value = 9}; last = None}
```

```ml
# dlst.last = None;;
- : bool = true
```

but ```dlst.last``` is still None!, when it should be pointing to same node as ```dlst.first```.

Similarly for ```insert_last```, ```first``` field remains ```None``` when input is _empyt doubly-linked list_.

https://github.com/cs3110/textbook-solutions/blob/80d164a4c0121cf6559b0880277ab255bc44c72a/chapter7.ml#L182-L187

**Comment by ishwar00** [comment]
bugs are resolved in PR #3

**Comment by clarksmr** [comment]
Mitigated by b91186d
